### PR TITLE
Azure hybrid use benefit linux

### PR DESCRIPTION
**Pull request by charmeling**
### Description

Identifying Linux virtual machines eligible for Azure Hybrid Use Benefit (AHUB) and automatically applying it 

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [X] All tests pass.
- [ ] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/hybrid_use_benefit_linux/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changelog
+
+## v1.0
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit_linux/CHANGELOG.md:3]: should be v2.0

File: cost/azure/hybrid_use_benefit_linux/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Azure Hybrid Use Benefit Policy
+
+## What it does
+
+Since November 2020 Microsoft also offers AHUB for certain Linux instances. This Policy Template is used to automatically apply the Azure Hybrid Use Benefit (AHUB) to all eligible Linux VMs in an Azure Subscription.
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit_linux/README.md:5]: The first sentence isn't necessary.

File: cost/azure/hybrid_use_benefit_linux/ahub_linux.pt [hunk 3]
```diff
@@ -0,0 +1,310 @@
+name "Azure Hybrid Use Benefit for Linux Server 1.0"
+rs_pt_ver 20180301
+type "policy"
+short_description "Identifies Linux instances eligible for Azure Hybrid Use Benefit.  See the [README](https://github.com/rightscale/policy_templates/tree/master/cost/azure/hybrid_use_benefit) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
+long_description ""
+severity "low"
+category "Cost"
+info(
+  version: "1.0",
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit_linux/ahub_linux.pt:9]: should be version 2.0

File: cost/azure/hybrid_use_benefit_linux/ahub_linux.pt [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  provider: "Azure",
+  service: "",
```
Comment (cdwilhelm) [inline cost/azure/hybrid_use_benefit_linux/ahub_linux.pt:11]: What is the service, Compute?

**Review by cdwilhelm — DISMISSED** [review]

File: cost/azure/hybrid_use_benefit_linux/CHANGELOG.md (hunk 1 above)
Comment (charmeling) [reply cost/azure/hybrid_use_benefit_linux/CHANGELOG.md:3]: Applied changes

**Comment by charmeling** [comment]
> This looks great @charmeling!
> Do you have a running example I could take a look at?
@srpomeroy - yes I tested it in the ThierryCorp org, not sure if you have access to it?
> Can you tell me what happens if you try to apply SLES AHUB to a RHEL VM and vice versa?
You got me ;-)
It depends on the distribution. 
For RHEL (applying SLES AHUB): license_type is set to "SLES_BYOS" but nothing really happens (if I check in the Azure console, no changes are shown.
For SLES (applying RHEL AHUB): error message in Postman that the action cannot be completed because Red Hat Cloud Access hasn't been enabled on my Azure subscription. However, if this would be the case, I would expect that similar to RHEL the wrong license_type would be set...
For SLES (applying SLES AHUB): it works - license_tpye=SLES_BYOS, also reflected in the Azure Console

I imagine, I need to define allowed values for both escalations, right?

**Comment by srpomeroy** [comment]
I am thinking we may only need a single escalation with logic that checks for the appropriate distro and applies the relevant BYOL string.

**Comment by charmeling** [comment]
@srpomeroy - Yes, that's a good idea, will see what I can do and update you.

**Comment by charmeling** [comment]
@srpomeroy - Yes, that's a good idea, I assume the check needs to be done in the js part, right? Or can I also do checks within the policy, escalation, workflow definitions? Will see what I can do...

**Comment by charmeling** [comment]
@srpomeroy - Did change it a bit. Now it checks what distribution is being used and only needs one single escalation. I could also verify in the CWF console in Rightscale that when doing the workflow part with a RHEL VM, it complains to activate cloud access in RedHat (see attachment)...
rhel_message0

**Review by charmeling — COMMENTED** [review]
Applied changes as requested

File: cost/azure/hybrid_use_benefit_linux/ahub_linux.pt [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+end
+
+escalation "license_instances" do
+  automatic contains($param_automatic_action, "Hybrid Use Benefit for Linux")
+  label "Approve AHUB for Linux"
+  description "Approve escalation to apply Hybrid Use Benefit to identified instances"
+  run "license_instances", data
+end
+
+#escalation "license_rhel_instances" do
```
Comment (srpomeroy) [inline cost/azure/hybrid_use_benefit_linux/ahub_linux.pt:252]: If we are good without this escalation block, let's go ahead and delete it.

**Review by srpomeroy — CHANGES_REQUESTED** [review]
Just a few minor tweaks.

**Review by srpomeroy — APPROVED** [review]
LGTM. Please just do me a favor @charmeling and fill out the initial post of this PR with information about the policy for reference.

**Comment by srpomeroy** [comment]
Any final remarks @cdwilhelm ?

**Comment by cdwilhelm** [comment]
@charmeling @srpomeroy nothing more from me. 
good work @charmeling

**Review by cdwilhelm — APPROVED** [review]
LGTM

**Pull request merged**